crates/bevy_ui/src/ui_node.rs

```diff
@@ -16,7 +16,15 @@ use std::ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Sub, SubAssign};
 #[reflect(Component, Default)]
 pub struct Node {
     /// The size of the node as width and height in pixels
-    pub size: Vec2,
+    ///
+    /// This Field should only read by the user
+    pub(crate) calculated_size: Vec2,
+}
+
+impl Node {
+    pub fn calculated_size(&self) -> &Vec2 {
```

Comment:
I _think_ that for math types of smaller size, more precisely when they fit in a single SIMD register, it's preferable to return by value. In general for types of say 16 Bytes of less it's faster to pass by value, as a rule of thumb.

Reply:
what do you mean? maybe make a method to get the desired value?

Reply:
I mean it should be in my opinion:

```suggestion
    pub fn calculated_size(&self) -> Vec2 {
```

Reply:
Why this doesn't create errors with the borrower??. I'm almost new at rust and i don't get it

Reply:
All variable bindings have 'move semantics',  which means that variables are moved by value, and previous variables holding that value cannot be used anymore. For example:

```rust
let x = String::new();

let y = x; // Here, we move `x` into `y`

// After the binding, `x` cannot be used anymore,
// as it has been moved to `y`.
//
// To make `x` and `y` both usable,
// you will have to clone it. (As long as the type implements `Clone`)
let x = String::new();

let y = x.clone(); // `x` is cloned, and its cloned value is moved to `y`

// Since `Clone::clone` takes `&self` and `self`,
// it merely borrows it and does not consume self.
// (i.e. it does not move into the `clone` method)

// An exception to this are `Copy` types, which means
// these types are implicitly duplicated by copying their bits.
// This is okay if the type is small, memory-wise.
let x = 5;

let y = x; // Integers implements `Copy`.
```

[`Vec2`] implements `Copy`, which means you don't have to explicitly clone the values. 

Further readings can be found here:

- [`Copy`]
- [`Clone`]
- [Understanding Ownership]

[`Vec2`]: https://docs.rs/glam/latest/glam/f32/struct.Vec2.html
[`Copy`]: https://doc.rust-lang.org/nightly/core/marker/trait.Copy.html
[`Clone`]: https://doc.rust-lang.org/nightly/core/clone/trait.Clone.html
[Understanding Ownership]: https://doc.rust-lang.org/book/ch04-00-understanding-ownership.html

Reply:
Ok, thanks!